ionc/ion_writer.c

```diff
@@ -1094,7 +1096,10 @@ iERR _ion_writer_add_annotation_helper(ION_WRITER *pwriter, ION_STRING *annotati
     ASSERT(annotation->length >= 0);
 
     if (!pwriter->annotations) {
-        IONCHECK(_ion_writer_set_max_annotation_count_helper(pwriter, DEFAULT_ANNOTATION_LIMIT));
+        int max_annotation_count = pwriter->options.max_annotation_count;
+        int final_max_annotation_count = (max_annotation_count > DEFAULT_ANNOTATION_LIMIT) ?
+                    max_annotation_count : DEFAULT_ANNOTATION_LIMIT;
```

Comment:
Let's just combine this into a single line; no need to store the `max_annotation_count` variable.

Reply:
Addressed.